### ISSUE TITLE
Parse Server crash on FCM response GOAWAY

### ISSUE THREAD

**Issue by mtrezza** [parse-community/parse-server-push-adapter#340]
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server-push-adapter/security/policy).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/main/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server-push-adapter/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the latest versions of [Parse Server](https://github.com/parse-community/parse-server/releases) and the [Parse Server Push Adapter](https://github.com/parse-community/parse-server-push-adapter/releases). 

### Issue Description

Parse Server crashes when FCM server responds with `GOAWAY`:

```
An uncaught exception occurred:
Error while making requests: GOAWAY - server_shutting_down, Error code: 0
Stack Trace:
  at ClientHttp2Session.<anonymous> (/var/app/current/node_modules/firebase-admin/lib/utils/api-request.js:994:23)
  at ClientHttp2Session.emit (node:events:517:28)
  at ClientHttp2Session.emit (node:domain:489:12)
  at Http2Session.onGoawayData (node:internal/http2/core:677:11)
```

These errors started to appear end of Nov 2024 without a change in the push adapter version, so there has likely been a change in the FCM server behavior. However, even if the firebase-admin SDK doesn't properly handle that response, the adapter should catch this error.

From a quick look into the adapter sending logic it seems that an error should be caught if the promise is rejected:

https://github.com/parse-community/parse-server-push-adapter/blob/5409e481d9708354be1cb453a3b481bb13d64e8a/src/FCM.js#L141-L145

However, it seems that the FCM client has a bug and doesn't handle the `GOAWAY` response properly, i.e. throwing an error instead of rejecting a promise, so that exception propagates all the way to the server. See also https://github.com/firebase/firebase-admin-node/issues/2789.

A workaround until the firebase-admin SDK is fixed could be to wrap this into a try/catch block.

### Steps to reproduce

This seems to happen when too many http2 connections are opened to the FCM server. There may be an additional underlying issue.

### Actual Outcome

Parse Server crashes.

### Expected Outcome

Parse Server should handle the error gracefully, log an error and not crash.

###  Environment

Client
- Parse Server Push Adapter version: `6.7.0`

Server
- Parse Server version: `7.3.0`

**Comment by mtrezza** [comment]
Probably related to https://github.com/firebase/firebase-admin-node/issues/2789

**Comment by siakc** [comment]
Please report your firebase-admin version in /var/app/current/node_modules/firebase-admin/package.json.

**Comment by mtrezza** [comment]
This has been observed with:

"firebase-admin": "12.5.0"
"firebase-admin": "13.0.1"

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.8.0](https://github.com/parse-community/parse-server-push-adapter/releases/tag/6.8.0)
